elide-async/src/main/java/com/yahoo/elide/async/models/security/AsyncAPIInlineChecks.java

```diff
@@ -18,51 +18,63 @@
 import com.yahoo.elide.core.security.checks.FilterExpressionCheck;
 import com.yahoo.elide.core.security.checks.OperationCheck;
 import com.yahoo.elide.core.security.checks.UserCheck;
+import com.yahoo.elide.core.type.ClassType;
 import com.yahoo.elide.core.type.Type;
 
 import java.security.Principal;
 import java.util.Collections;
 import java.util.Optional;
 
 /**
- * Operation Checks on the Async Query and Result objects.
+ * Operation Checks on the Async API and Result objects.
  */
-public class AsyncQueryInlineChecks {
+public class AsyncAPIInlineChecks {
     private static final String PRINCIPAL_NAME = "principalName";
 
-    static private FilterPredicate getPredicateOfPrincipalName(String principalName) {
-        Path.PathElement path = new Path.PathElement(AsyncQuery.class, String.class, PRINCIPAL_NAME);
+    static private FilterPredicate getPredicateOfPrincipalName(String principalName, Class<?> clazz) {
```

Comment:
You can build filter predicates using Type (instead of class.  Everywhere in this file you see Class<?> change it to Type<?>.